packages/components/MenuButton/src/MenuButton.types.ts

```diff
@@ -0,0 +1,40 @@
+import { ContextualMenuItemProps, ContextualMenuProps } from '@fluentui-react-native/contextual-menu';
+import { IButtonProps } from '@fluentui-react-native/button';
+import { IRenderData } from '@uifabricshared/foundation-composable';
+
+export const MenuButtonName = 'MenuButton';
+
+export interface MenuButtonContext {
+  showContextualMenu?: boolean;
+}
+
+export interface MenuButtonState {
+  context: MenuButtonContext;
+}
+
+export interface MenuButtonItemProps extends ContextualMenuItemProps {
+  submenu?: boolean;
+  submenuItems?: ContextualMenuItemProps[];
+}
+
+export interface MenuButtonProps extends IButtonProps {
+  menuItems?: MenuButtonItemProps[];
+  onItemClick?: (key: string) => void;
+  contextualMenu?: ContextualMenuProps;
+}
+
+export type MenuButtonSlotProps = {
+  root: MenuButtonProps;
+  button: IButtonProps;
+  contextualMenu: React.PropsWithRef<ContextualMenuProps>;
+  contextualMenuItems: Pick<MenuButtonProps, 'menuItems'>;
+  contextualMenuItem: MenuButtonItemProps;
```

Comment:
Are the latter 2 slots not picked up as "Sub slots" of the `contextualMenu` slot? Does it not work that way hence needing to specify it twice?

Reply:
ContextualMenu has items as children. contextualMenuItems slot was added to wrap SubmenuItems. Because it should look like:
```<ContextualMenu>
  <>
  <SubmenuItem />
  <Submenu>
         <ContextualMenuItem/>
  </Submenu>
  </>
</ContextualMenu>
```
I wanted not to use slots for React.Fragment wrapper, but I got an error. I added `/** @jsxFrag React.createFragment */` but it didn't help.

Reply:
Ah.. fragments. I have no idea how that works with slots. That's actually very interesting, I'll look more into that!